### ISSUE TITLE
bug in test mechanism

### ISSUE THREAD

**Issue by kolipka** [Samsung/iotjs#1090]
I am trying to check IoT.js test suite on TizenRT. In parallel suite I discovered that our testing mechanism (check_test.js / test_runner.js) have some flows. I receive status that test-http-status-message.js failed, but in fact problem is in test-http-catch-uncaughtexception.js. You can see the same behaviour on linux, when you change [test-http-catch-uncaughtexception.js](https://github.com/Samsung/iotjs/blob/master/test/node/parallel/test-http-catch-uncaughtexception.js#L44) line to: `assert.strictEqual(er.message, 'get did fail__');`. On TizenRT i have _'socket hang up'_ so exactly this assert fails.

To be honest: I have no idea what to do with this. It seems that IoT.js tests are not realiable, and this is quite scary. I understand that this is special case of async test and exception, but it makes debugging errors more hard.

I am not sure what is the purpose of parallel suite. I see that it is copied from node.js repository, but I suspect their test runner is pretty differend.

**Comment by LaszloLango** [comment]
Thank you for your kind input. We have (already) taken into account your inconvenience and that is why there is the remote test runner and the previously discontinued python test runner. Please see to the other Issues/PRs for further info. https://github.com/Samsung/iotjs/pull/655#issuecomment-283678236
IMHO, we should reopen the #655.

**Comment by glistening** [comment]
@kolipka, @LaszloLango We've discontinued python test runner with JS test runner in hope that JS test runner will cover low end device/os like stm32f4-discovery/nuttx, which lacks of python. But till now, we've got continuous report of limitation of JS test runner and we have and can use remote test runner in stm32f4-discovery/nuttx. Furthermore, remote test runner on artik053/tizenrt is almost done with serial communication. So we are thinking about restoring python test driver for linux on {desktop or RPI}.
